### PR TITLE
Aggregator extraVolumes and extraVolumeMounts

### PR DESCRIPTION
**Pull request by thomasvn**
## What does this PR change?

- Allows users to add `extraVolumes` and `extraVolumeMounts` to the Aggregator pod and CloudCost pod, when Aggregator is running as a Statefulset.
- Cleanup values.yaml file under `kubecostAggregator`

## Does this PR rely on any other PRs?

No

## How does this PR impact users? (This is the kind of thing that goes in release notes!)

As titled

## Links to Issues or tickets this PR addresses or fixes

Implements https://github.com/kubecost/cost-analyzer-helm-chart/issues/3261

## What risks are associated with merging this PR? What is required to fully test this PR?

None. This is an optional config.

## How was this PR tested?

```yaml
kubecostAggregator:
  deployMethod: "statefulset"
  extraVolumes:
    - name: test1
      secret:
        secretName: test1-secret
  extraVolumeMounts:
    - name: test1
      mountPath: /var/configs/test1
  cloudCost:
    extraVolumes:
      - name: test2
        configMap:
          name: test2-configmap
    extraVolumeMounts:
      - name: test2
        mountPath: /var/configs/test2
kubecostModel:
  federatedStorageConfigSecret: "federated-store"
```

```sh
helm template ./cost-analyzer -f values.yaml
```

Validated that the `test1` and `test2` volumes and volumemounts were templated as expected.

## Have you made an update to documentation? If so, please provide the corresponding PR.

Documented via values.yaml

### CODE REVIEW

File: cost-analyzer/values.yaml [hunk 1]
```diff
@@ -2450,64 +2450,50 @@ kubecostAggregator:
...
-  #   runAsNonRoot: true
-  # containerSecurityContext:
-  #   allowPrivilegeEscalation: false
-  #   readOnlyRootFilesystem: true
-  #   runAsNonRoot: true
-  #   seccompProfile:
-  #     type: RuntimeDefault
-  #   capabilities:
-  #     drop:
-  #       - ALL
```
Comment (thomasvn) [inline cost-analyzer/values.yaml:2488]: Deleting this comment because we already have `securityContext` and `containerSecurityContext` commented below.

**Comment by thomasvn** [comment]
@ravnalexquinn @Kavinraja-Ganesan let me know what you think!

**Comment by Kavinraja-Ganesan** [comment]
@thomasvn Can we please include `extraEnvs` as well? Others lg2m!

**Comment by thomasvn** [comment]
@Kavinraja-Ganesan `extraEnvs` for Aggregator pod and CloudCost pod should already be available! It would look something like this:

```yaml
kubecostAggregator:
  env:
    "key": "value"
  cloudCost:
    env:
      "key2": "value2"
```

**Comment by Kavinraja-Ganesan** [comment]
Thanks @thomasvn! Subscribed and waiting for this release.

**Comment by Kavinraja-Ganesan** [comment]
Can this be included in v2.2.1? (if possible)

**Review by JesseGoodier — APPROVED** [review]

**Comment by ravnalexquinn** [comment]
Hi @thomasvn, thanks for this

Regarding this:
`Allows users to add extraVolumes [...] when Aggregator is running as a Statefulset.`

Could you explain why the aggregator running as a stateful set is a requirement, please? Preventing the mount values being referenced when aggregator is a container running under the 'cost-analyzer' pod

**Comment by thomasvn** [comment]
@ravnalexquinn The original thinking, was that when all containers are running under the `cost-analyzer` pod, they all reference the `persistent-configs` volumeMount. ([here](https://github.com/kubecost/cost-analyzer-helm-chart/blob/7b6e654d96390fefd72dbda75469541a7f524ab7/cost-analyzer/templates/_helpers.tpl#L913-L915) for aggregator, [here](https://github.com/kubecost/cost-analyzer-helm-chart/blob/7b6e654d96390fefd72dbda75469541a7f524ab7/cost-analyzer/templates/cost-analyzer-deployment-template.yaml#L553-L554) for cost-model).

In other words, when Aggregator is run in the `cost-analyzer` pod, both the Aggregator and Cost-model have read write access to the `/var/configs` directory. Therefore adding an `extraVolumeMount` to the cost-analyzer (so long as its under `/var/configs`) should also be accessible to the Aggregator. Does that fit your use case here?
